index.js

```diff
@@ -59,15 +60,26 @@ function sendMachines(res, token) {
     var promises = ids.map(function(machineId) {
       return redis.hgetall(machineId)
       .then(function(machine) {
-        machines[machineId] = machine;
+        return redis.hgetall(token + ':' + machineId)
+        .then(function(machineClients) {
+          machine.clients = machineClients;
+          machines[machineId] = machine;
+        });
       });
     });
 
+    promises.push(
+        redis.smembers(token + ':clients')
```

Comment:
Aren't you using token:clients only to build the table in the page?
You could create the clients array by using machine.clients